### ISSUE TITLE
How to implement git commit command

### ISSUE THREAD

**Issue by Rupangkan** [fleipold/jproc#19]
I am trying to write a code to commit with git but it is working sometimes and not working the other time. Here is the code for it:

```
ProcResult result = new ProcBuilder("git")
                .withArg("init")
                .withWorkingDirectory(new File(dir))
                .run();
```

        result = new ProcBuilder("git")
                .withArgs("add",".")
                .run();
        result = new ProcBuilder("git")
                .withArgs("commit","-m").withArg("1st commit")
                .run();
        System.out.println("Successfully commited");
I tried something like the given code below but it still doesn't work:

```
result = new ProcBuilder("git")
            .withArgs("commit","-m").withArg(""+(char)34+"1st commit"+(char)34+"")
            .run();
```
It throws the following exception:

Exception in thread "main" org.buildobjects.process.ExternalProcessFailureException: External process 'git commit -m '1st commit'' returned 1 after 126ms

    at org.buildobjects.process.Proc.<init>(Proc.java:126)
    at org.buildobjects.process.ProcBuilder.run(ProcBuilder.java:205)
    at ProjectInitializerAutomation.ProjectInitializerAutomation.Local.command(Local.java:90)
    at ProjectInitializerAutomation.ProjectInitializerAutomation.Local.createFolder(Local.java:58)
    at ProjectInitializerAutomation.ProjectInitializerAutomation.Local.create(Local.java:39)
    at ProjectInitializerAutomation.ProjectInitializerAutomation.Driver.main(Driver.java:21)``

**Comment by fleipold** [comment]
This means that the git command returned a non zero exit code (1). Interestingly there is no error output (this would normally be included in the exception.
Unfortunately standard out is not captured in the exception. You can suppress the exception with:
~~~.java
result = new ProcBuilder("git")
            .withArgs("commit","-m").withArg(""+(char)34+"1st commit"+(char)34+"")
            .withExpectedExitStatuses(0, 1)
            .run();
~~~

If you then print the output like so:

~~~.java
System.out.println(result.getOutputString());
//now bail
if (result.getExitValue()!=0)
  throw new RuntimeException("Failed to run git");
~~~

You might see what is going wrong. 

Also, you don't need to add the quotation marks to the argument. they normally tell bash that your message is a single argument, i.e. they are stripped off when the arguments are passed to the process.

**Comment by Rupangkan** [comment]
@fleipold Thank you, I found out the flaw in my code. Your suggestion was beneficial.

**Comment by fleipold** [comment]
@Rupangkan I am glad you figured it out. Thank you for this bug report. I have improved the exception message, so it now includes standard out from the failed command (available in the latest release 2.5.1), here an example:
~~~
Exception in thread "main" org.buildobjects.process.ExternalProcessFailureException: External process `git` terminated with unexpected exit status 1 after 39ms:
  $ git commit -m '1st commit'
  STDOUT: On branch master
  STDOUT:
  STDOUT: Initial commit
  STDOUT:
  STDOUT: nothing to commit (create/copy files and use "git add" to track)
  ~~~
